test/extension.jl

```diff
@@ -8,9 +8,10 @@ struct PolyTerm <: AbstractTerm
     deg::Int
 end
 PolyTerm(t::Term, deg::ConstantTerm) = PolyTerm(t.sym, deg.n)
+poly(t::AbstractTerm, deg) = PolyTerm(t, deg)
 
-StatsModels.apply_schema(t::FunctionTerm{typeof(poly)}, sch, ::Type{<:PolyModel}) =
-    PolyTerm(t.args_parsed...)
+StatsModels.apply_schema(t::FunctionTerm{typeof(poly)}, sch::Schema, Mod::Type{<:PolyModel}) =
+    apply_schema(poly(t.args...), sch, Mod)
```

Comment:
Here too, what's going to happen if a package uses a similar pattern to define a custom term?

Reply:
same as always, you'll get method invalidation errors and/or the context will shuttle dispatch in the right way in the right model contexts

Reply:
method invalidation *warnings, not errors